### PR TITLE
Categorical count

### LINKED ISSUE #97: Count datafield
## Problem
I've noticed that draco sometimes recommends the datafield "Count of records" to a variable. I think this is only applicable to categorical data. When editing the chart, this appears as a variable without a datafield, but with a count aggregation. This is a little confusing, and it also means that the user cannot currently deliberately add this kind of count variable.

Screen Shot 2021-08-10 at 2 33 26 PM
Screen Shot 2021-08-10 at 2 33 41 PM

## Options
- "Count of Records" could be added as a new datafield list entry. It might make sense to always have it at the top of the list.
- We could also allow for aggregation (including count) of categorical variables.

### PR DESCRIPTION
**Pull request by jahn96**
Fix #97 
* Allowed for count aggregation of categorical variables.

## Future works
* Changing the categorical variable to quantitative variable when the count aggregation is applied
* Support groupby

### CODE REVIEW

File: src/components/Onboarding/ColumnSelectorSidebar.tsx [hunk 1]
```diff
@@ -146,6 +147,22 @@ export default function ColumnSelectorSidebar(props: { plotArgs: Args }) {
...
+    return produce(spec, (gs) => {
+      if ('aggregate' in gs.encoding.x && !('field' in gs.encoding.x)) {
+        (gs.encoding.x as any).type = 'nominal';
+        (gs.encoding.x as any).field = (gs.encoding?.y as any).field;
+      } else if ('aggregate' in gs.encoding.y && !('field' in gs.encoding.y)) {
+        (gs.encoding.y as any).type = 'nominal';
+        (gs.encoding.y as any).field = (gs.encoding.x as any).field;
+      }
+    });
+  }
```
Comment (Waidhoferj) [inline src/components/Onboarding/ColumnSelectorSidebar.tsx:164]: Not sure I understand why we are flip flopping the axes here... I think we can just apply the count as a transformation. In the Vega docs, it says that a count on one field applies to all fields, so you can use that to your advantage to assign count to a hidden filter.

File: src/components/Sidebar/Tabs/FilterScreen.tsx [hunk 2]
```diff
@@ -381,6 +390,17 @@ function CategoricalFilters(props: FilterGroupProps) {
           </li>
         ))}
       </ul>
+      <article className={'aggregation-article'}>
+        <h3>Transformation</h3>
```
Comment (Waidhoferj) [inline src/components/Sidebar/Tabs/FilterScreen.tsx:394]: "Aggregation"

**Review by Waidhoferj — CHANGES_REQUESTED** [review]
I noticed that assigning count crashes the chart in some circumstances:
https://user-images.githubusercontent.com/33008436/130290164-e1e026dd-1030-4c72-85ee-740511b6f2c2.mov

Also there is an issue with sampling here that doesn't exist on the main branch:
Screen Shot 2021-08-20 at 1 22 04 PM

**Comment by Waidhoferj** [comment]
> Also there is an issue with sampling here that doesn't exist on the main branch:
> Screen Shot 2021-08-20 at 1 22 04 PM

I think this was just an issue with my jupyter-bifrost-tracing

File: src/components/Onboarding/ColumnSelectorSidebar.tsx (hunk 1 above)
Comment (jahn96) [reply src/components/Onboarding/ColumnSelectorSidebar.tsx]: Removed this

**Review by Waidhoferj — APPROVED** [review]
LGTM

**Pull request merged**